src/plugins/intel_gpu/tests/test_cases/concatenation_gpu_test.cpp

```diff
@@ -599,9 +620,101 @@ struct concat_gpu_4d : public concat_gpu {
     }
 };
 
+// Test case for axis=3 case in 4D
+template <typename Type>
+struct concat_gpu_4d_axis3 : public concat_axis3_gpu {
+public:
+
+    void test(format::type fmt) {
+        auto data_type = type_to_data_type<Type>::value;
+
+        auto& engine = get_test_engine();
+        const size_t batch_num = testing::get<0>(GetParam());
+        const size_t in_feature = testing::get<1>(GetParam());
+        const size_t input_y = testing::get<2>(GetParam());
+        const std::vector<size_t> input_x = testing::get<3>(GetParam());
+        size_t output_x = 0;
+        for (auto& x : input_x)
+            output_x += x;
+
+        topology topology;
+
+        std::vector<VVVVF<Type>> in_data;
+        std::vector<memory::ptr> in_memory;
+        std::vector<primitive_id> input_ids;
+        for (size_t i = 0; i < input_x.size(); i++) {
+            auto size = tensor(static_cast<int32_t>(batch_num),
+                               static_cast<int32_t>(in_feature),
+                               static_cast<int32_t>(input_x[i]),
+                               static_cast<int32_t>(input_y));
+            auto data = generate_random_4d<Type>(batch_num, in_feature, input_y, input_x[i], -1, 1);
+            auto in_lay = layout(data_type, fmt, size);
+            auto data_flat = std::vector<Type>(in_lay.get_linear_size(), 0);
+
+            for (size_t bi = 0; bi < batch_num; ++bi) {
```

Comment:
This part seems common with original 4d tensor. Could we extract common part of blocked format data generation as a function?

Reply:
It is also weird that we dont have tests for all axes. Lets add them in the following up tasks!